### PR TITLE
add NEAR DID method(#68)

### PR DESCRIPTION
**Pull request by OntologyTech**
We have developed this specification to define a new decentralized identifier (DID) method for hosting DIDs on the NEAR blockchain, also referred to as NEAR DID, and facilitate developers to work with related contracts.

A reference implementation of NEAR DID scheme is currently under development.

### CODE REVIEW

File: text/0000-near-did-method.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+The public key of the Near account's access key that was used to sign the transaction will be inserted into the `publicKey` property in the valid form and will be referred to in the `authentication` property.
+
+This method will return true if the new NEAR DID is registered successfully.
+
+## Read (Resolve)
+[read]: #read
+
+NEAR DID's associated DID document can be looked up by invoking the `getDocument` method of the registry.
+
+To ensure the smart contract invocation result is trustworthy, the client could query a certain number of nodes and then compare the return values.
```
Comment (bowenwang1996) [inline text/0000-near-did-method.md:76]: I don't think it's necessary. If the client doesn't trust some public rpc server they can use their own.

**Comment by OntologyTech** [comment]
> Two questions:
> 
> * What is the specification of the DID document? How large can it be?

Link to the NEAR DID document specification: https://github.com/ontology-tech/DID-spec-near/blob/master/NEAR/NEAR-DID-spec.md
The reference implementation is available here: https://github.com/ontology-tech/DID-NEAR-rust
Generally speaking, the size of a NEAR DID document depends on the number of the bound verification methods. A greater number of verification methods would imply a larger document size.

> * How does composability work? Can you give some examples of how a DID is used?
 
Could you describe in what context you are referring to 'composability'? 
For instance, on Ontology, the DID framework ONT ID serves to support the account system that can be used to identify individuals and entities and record and validate their actions on-chain. There is value in establishing this distinction from traditional wallet addresses, since this allows for greater logical clarity when it comes to more complex operations as traditional wallet accounts are normally linked to assets and asset related operations. 
A more specific implementation would be something like issuing and verifying credentials for entities which could denote a whole variety of things based on the subject and scenario.

W3C lays out certain potential use cases for distributed identifiers. Follow this link for more details: https://www.w3.org/TR/did-use-cases/

File: text/0000-near-did-method.md (hunk 1 above)
Comment (OntologyTech) [reply text/0000-near-did-method.md:76]: > I don't think it's necessary. If the client doesn't trust some public rpc server they can use their own.

Sure, this definitely isn't a necessity, just a possible solution. Clients may or may not run their own server. It is up to them to use the server of their choice.

**Comment by bowenwang1996** [comment]
Maybe I missed something, but it seems that the question of authorization is not well explained. It seems that the contract will be deployed on every account that participates in the system, so what prevents the account owner from modifying the data arbitrarily (the document and service endpoint for example)? Or is there some incentive that would prevent them from doing so?

**Comment by OntologyTech** [comment]
> Maybe I missed something, but it seems that the question of authorization is not well explained. It seems that the contract will be deployed on every account that participates in the system, so what prevents the account owner from modifying the data arbitrarily (the document and service endpoint for example)? Or is there some incentive that would prevent them from doing so?

Good question. So the thing is, a single contract is deployed on the chain, and not one for each user that is part of the system. A new document is generated in the contract for each new DID. DID subjects can authorize other DID subjects to update their DID documents by assigning the `controller` property. Whether an owner can perform such arbitrary or malicious actions relates to  governance model specifics.

**Comment by bowenwang1996** [comment]
> So the thing is, a single contract is deployed on the chain

I see. But it doesn't look like the case from your contract implementation. I suppose the contract should store the DIDs?

**Comment by OntologyTech** [comment]
That is correct. The DID documents for each ID instance are stored within the contract. 
You can refer to the implementation here for better clarity: https://github.com/ontology-tech/DID-NEAR-rust

**Comment by frol** [comment]
@OntologyTech Do you still have a use-case for this proposal? Would you like to work on it and shape it together?

**Comment by ori-near** [comment]
As the moderator, I assume this PR is no longer active given that there was no activity for over two months. Therefore, I am closing it. If anyone is interested in reopening this PR, please submit a new one.

**Pull request closed without merging**